File: 100sameTree/sameTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function(p, q) {
    return p && q ? Object.is(p.val, q.val) && isSameTree(p.left, q.left) && isSameTree(p.right, q.right) : Object.is(p, q);
};

File: 100sameTree/sameTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        return Objects.nonNull(p) && Objects.nonNull(q) ? p.val == q.val && this.isSameTree(p.left, q.left) && this.isSameTree(p.right, q.right) : p == q;
    }
}
